internal/xds/rbac/rbac_engine_test.go

```diff
@@ -0,0 +1,625 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rbac
+
+import (
+	"testing"
+
+	v3corepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	v3rbacpb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	v3routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	v3matcherpb "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	v3typepb "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	wrapperspb "github.com/golang/protobuf/ptypes/wrappers"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/peer"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+type addr struct {
+	ipAddress string
+}
+
+func (addr) Network() string   { return "" }
+func (a *addr) String() string { return a.ipAddress }
+
+// TestRBACEngineConstruction tests the construction of the RBAC Engine. Due to
+// some types of RBAC configuration being logically wrong and returning an error
+// rather than successfully constructing the RBAC Engine, this test tests both
+// RBAC Configurations deemed successful and also RBAC Configurations that will
+// raise errors.
+func (s) TestRBACEngineConstruction(t *testing.T) {
+	tests := []struct {
+		name                  string
+		rbacConfig            *v3rbacpb.RBAC
+		successfullyConstruct bool
```

Comment:
This can be restructed as `wantErr bool`.
This is a common idiom in Go tests where you call a function and test whether it returned an error or not.
Also, the zero value for a boolean field is false, so you can skip setting this value to false explicitly in the test cases.

Reply:
Deleted explicit setting to false in second test as well.